src/prefect/orion/orchestration/core_policy.py

```diff
@@ -406,13 +442,20 @@ class PreventTransitionsFromTerminalStates(BaseOrchestrationRule):
     FROM_STATES = TERMINAL_STATES
     TO_STATES = ALL_ORCHESTRATION_STATES
 
+    NAMED_STATE_EXCEPTIONS = {
+        "Restarting",
```

Comment:
Is this the name that's used for RUNNING on a restart? It mirrors "Retrying" but is unfortunately less clear and feels easily confused with the scheduled state. Perhaps @billpalombi has a suggestion here.

I think that AwaitingRetry -> Retrying makes sense for retries but for restarts the action isn't automated so it feels more like Restarting -> Running?

Reply:
I think:
- the named SCHEDULED equivalent should be "AwaitingRestart" to mirror "AwaitingRetry"
- the named RUNNING equivalent should be "Rerunning." I don't love it either, but I can't think of anything better.

Reply:
I don't have an opinion about the named state here--I think `Retrying` makes just as much sense as anything else

Reply:
Do IU understand correctly that retry counts are zeroed out when a flow run is restarted? If so, my only concern with `Retrying` is that a user might expect that it counts toward their retry count, but perhaps we don't need to signal that with the state name.

Reply:
Yeah the retry counts are zeroed so it can do its full set of retries again. 

> the named SCHEDULED equivalent should be "AwaitingRestart" to mirror "AwaitingRetry"

The problem with this is that it makes it seem like you're waiting for someone to click the restart button but this is actually what happens right after you click it. I'm not super opposed though.

> the named RUNNING equivalent should be "Rerunning." I don't love it either, but I can't think of anything better.

Can we just use "Running" as a name and consider a custom name later? A custom name was necessary for Dustin's original implementation but it should not be now. I do not think we necessarily need to try to clarify this as we might as well see if using the normal name is sufficient.

Reply:
If we don't need custom names, I agree that they may only create confusion. For example, if my flow run retries after having been restarted, we'll enter the usual state names and the indication that the flow run had been restarted will be lost. I'm on board with using "Scheduled" and "Running" as usual.